lab2.js

```diff
@@ -118,33 +166,56 @@ assert((new Human()).sayHello(new Klingon()) === 'nuqneH',
 // will test your code)
 //*********************************************************
 
+var cities = ['Chicago', 'Paris', 'Los Angeles', 'Vancouver'];
+var numbers1 = [1, 2, 3, 4, 5];
+var numbers2 = [10, 15, 20, 25, 30];
+var numbers3 = [37, 1, 12, 70, 18];
+var allNumbers = [numbers1, numbers2, numbers3];
+
 function lastLetterSort(stringArray) {
+
   function byLastLetter(a, b) {
-    //TODO: implement me. sort the strings in alphabetical
-    // order using their last letter
-    // Read this about how the sort function works:
-    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
-    // this byLastLetter function is a "compare function"
-    // And check out the "comparing strings" section  here:
-    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String
+
+    if ((a.charAt(a.length - 1)) < (b.charAt(b.length - 1))) {
+      return -1;
+    } else if ((a.charAt(a.length - 1)) > (b.charAt(b.length - 1))) {
+      return 1;
+    }
+    return 0;
   }
   stringArray.sort(byLastLetter);
 }
 
 function sumArray(numberArray) {
+
   var sum = 0;
-  // TODO: implement me using forEach
+
+  numberArray.forEach(function(value) {
+    sum += value;
+  });
   return sum;
 }
 
 function sumSort(arrayOfArrays) {
-  arrayOfArrays.sort(function(item) {
-    // TODO: implement me using sumArray
-    //  order the arrays based on the sum of the numbers
-    //  inside each array
+  var sumSortArray = [];
+
+  arrayOfArrays.forEach(function(array) {
+    sumSortArray.push(sumArray(array));
   });
+
+  arrayOfArrays.sort(sumSortArray);
```

Comment:
if you're looking to sort sumSortArray, you can just do this:
sumSortArray.sort()